src/OrchardCore/OrchardCore.Modules/ModularTenantContainerMiddleware.cs

```diff
@@ -40,17 +40,17 @@ public class ModularTenantContainerMiddleware
             {
                 var shellContext = _orchardHost.GetOrCreateShellContext(shellSetting);
 
-                var existingRequestServices = httpContext.RequestServices;
-
                 using (var scope = shellContext.EnterServiceScope())
                 {
-                    if (!shellContext.IsActivated)
+                    if (!shellContext.IsActivated || shellContext.IsActivating)
                     {
                         lock (shellContext)
```

Comment:
We can get here is `IsActivating` is true, and then waiting until it's activated, but then we should not try to activate it.
I don't see why we need `IsActivating` , instead I would just move `shellContext.IsActivated = true;` after the `Activated` events.